sound/soc/sof/ipc4-pcm.c

```diff
@@ -234,24 +292,43 @@ static int sof_ipc4_pcm_dai_link_fixup(struct snd_soc_pcm_runtime *rtd,
 	struct snd_interval *rate = hw_param_interval(params, SNDRV_PCM_HW_PARAM_RATE);
 	struct snd_mask *fmt = hw_param_mask(params, SNDRV_PCM_HW_PARAM_FORMAT);
 	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(component);
+	struct snd_soc_dai *cpu_dai = asoc_rtd_to_cpu(rtd, 0);
 	struct sof_ipc4_copier *ipc4_copier;
+	struct snd_soc_dapm_widget *w;
+	bool use_chain_dma = false;
+	int dir;
 
 	if (!dai) {
 		dev_err(component->dev, "%s: No DAI found with name %s\n", __func__,
 			rtd->dai_link->name);
 		return -EINVAL;
 	}
 
+	for_each_pcm_streams(dir) {
```

Comment:
nit-pick. can we add a comment saying we should not change format when using chained DMA

Reply:
The comment is there couple of lines down, where the format mask is set, but maybe it could be more explicit.